indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -365,6 +365,7 @@ public TaskStatus call()
                                   )
                               );
 
+                              command.add(String.format("-Ddruid.indexer.taskId=%s", task.getId()));
```

Comment:
proposal:

remove

 ```java
  public static void addResponseHeaderFilter(Binder binder, String path, Map<String, String> headers)
   {
     Multibinder.newSetBinder(binder, ServletFilterHolder.class)
                .addBinding()
                .toInstance(new JettyResponseHeaderFilterHolder(path, headers));
   }
```

And instead call 

```java
Multibinder.newSetBinder(binder, ServletFilterHolder.class)
                .addBinding()
                .to(ChatResponseHeaderFilter.class);
```
 

Then further in the module you can specify the `ChatResponseHeaderFilter` provider with the appropriate injected stuff.